components/popover/popover-mixin.js

```diff
@@ -127,68 +215,375 @@ export const PopoverMixin = superclass => class extends superclass {
+				<div></div>
+			</div>
+		` : nothing;
+
+		return html`${content}${pointer}`;
+
+	}
+
+	async resize() {
+		if (!this._opened) return;
+		await this.#position();
 	}
 
 	toggleOpen(applyFocus = true) {
 		if (this._opened) return this.close();
 		else return this.open(!this._noAutoFocus && applyFocus);
 	}
 
-	_addAutoCloseHandlers() {
-		this.addEventListener('blur', this._handleAutoCloseFocus, { capture: true });
-		document.body.addEventListener('focus', this._handleAutoCloseFocus, { capture: true });
-		document.addEventListener('click', this._handleAutoCloseClick, { capture: true });
+	#handleAncestorMutationBound;
+	#handleAutoCloseClickBound;
+	#handleAutoCloseFocusBound;
+	#handleResizeBound;
+	#repositionBound;
+
+	#addAutoCloseHandlers() {
+		this.addEventListener('blur', this.#handleAutoCloseFocusBound, { capture: true });
+		document.body.addEventListener('focus', this.#handleAutoCloseFocusBound, { capture: true });
+		document.addEventListener('click', this.#handleAutoCloseClickBound, { capture: true });
 	}
 
-	_clearDismissible() {
+	#addRepositionHandlers() {
+
+		const isScrollable = (node, prop) => {
+			const value = window.getComputedStyle(node, null).getPropertyValue(prop);
+			return (value === 'scroll' || value === 'auto');
+		};
+
+		this.#removeRepositionHandlers();
+
+		window.addEventListener('resize', this.#handleResizeBound);
+
+		this._ancestorMutationObserver ??= new MutationObserver(this.#handleAncestorMutationBound);
+		const mutationConfig = { attributes: true, childList: true, subtree: true };
+
+		let node = this;
+		this._scrollablesObserved = [];
+		while (node) {
+
+			// observe scrollables
+			let observeScrollable = false;
+			if (node.nodeType === Node.ELEMENT_NODE) {
+				observeScrollable = isScrollable(node, 'overflow-y') || isScrollable(node, 'overflow-x');
+			} else if (node.nodeType === Node.DOCUMENT_NODE) {
+				observeScrollable = true;
+			}
+			if (observeScrollable) {
+				this._scrollablesObserved.push(node);
+				node.addEventListener('scroll', this.#repositionBound);
+			}
+
+			// observe mutations on each DOM scope (excludes sibling scopes... can only do so much)
+			if (node.nodeType === Node.DOCUMENT_NODE || (node.nodeType === Node.DOCUMENT_FRAGMENT_NODE && node.host)) {
+				this._ancestorMutationObserver.observe(node, mutationConfig);
+			}
+
+			node = getComposedParent(node);
+		}
+
+		this._openerIntersectionObserver = new IntersectionObserver(entries => {
+			entries.forEach(entry => this._offscreen = !entry.isIntersecting);
+		}, { threshold: 0 }); // 0-1 (0 -> intersection requires any pixel visible, 1 -> intersection requires all pixels visible)
+		if (this._opener) {
+			this._openerIntersectionObserver.observe(this._opener);
+		}
+
+	}
+
+	#clearDismissible() {
 		if (!this._dismissibleId) return;
 		clearDismissible(this._dismissibleId);
 		this._dismissibleId = null;
 	}
 
-	_focusContent(container) {
+	#constrainSpaceAround(spaceAround, spaceRequired, openerRect) {
```

Comment:
This is basically adapted from `DropdownContentMixin`, except the `boundary` constraints aren't included here (yet), and `align` is changed to position `span` to align with what the browser is doing for positioning in the logical 3x3 grid.